common/src/JetIds.cxx

```diff
@@ -53,54 +53,229 @@ bool DeepCSVBTag::operator()(const Jet & jet, const Event &) const{
-    if(fabs(jet.eta())>2.7)
+  return false;
+}
+
+//PUPPI with puppi multiplicities, according to https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetID13TeVRun2017
+bool JetPFID::tightID2017_PUPPI(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.7
+     && jet.numberOfDaughters()>1 
+     && jet.neutralHadronEnergyFraction()<0.90
+     && jet.neutralEmEnergyFraction()<0.90){
+    
+    if(fabs(jet.eta())>=2.4)
       return true;
+    
+    if(jet.chargedHadronEnergyFraction()>0
+       && jet.chargedMultiplicity()>0)
+      return true;   
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  &&jet.neutralHadronEnergyFraction()<0.99){
+    return true;
+  }
+  else if(fabs(jet.eta())>3
+	  && jet.neutralPuppiMultiplicity()>2 && jet.neutralPuppiMultiplicity()<15
+	  && jet.neutralEmEnergyFraction()<0.90
+	  && jet.neutralHadronEnergyFraction()>0.02){
+    return true;
+  }
+  return false;
+}
+
+bool JetPFID::tightLepVetoID2017(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.7 && !tightID2017_CHS(jet)) return false;
+  if(jet.muonEnergyFraction() <0.8
+     &&jet.chargedEmEnergyFraction()<0.80)
+    return true;
+  return false;
+}
+
+//according to https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetID13TeVRun2018
+bool JetPFID::tightID2018_CHS(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.6
+     && jet.numberOfDaughters()>1 
+     && jet.neutralHadronEnergyFraction()<0.90
+     && jet.neutralEmEnergyFraction()<0.90
+     && jet.chargedHadronEnergyFraction()>0
+     && jet.chargedMultiplicity()>0){
+  }
+  else if(fabs(jet.eta())>2.6 && fabs(jet.eta())<=2.7
+	  && jet.neutralHadronEnergyFraction()<0.90 
+	  && jet.neutralEmEnergyFraction()<0.99
+	  && jet.chargedMultiplicity()>0){
+    return true;
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  &&jet.neutralEmEnergyFraction()<0.99
+	  &&jet.neutralEmEnergyFraction()>0.02
+	  &&jet.neutralMultiplicity()>2){
+    return true;
+  }
+  else if(fabs(jet.eta())>3
+	  && jet.neutralMultiplicity()>10
+	  && jet.neutralEmEnergyFraction()<0.90
+	  && jet.neutralHadronEnergyFraction()>0.02){
+    return true;
+  }
   return false;
 }
 
-bool JetPFID::tightLepVetoID(const Jet & jet) const{
-  if(!tightID(jet))return false;
+bool JetPFID::tightLepVetoID2018(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.6 && !tightID2018_CHS(jet)) return false;
   if(jet.muonEnergyFraction() <0.8
      &&jet.chargedEmEnergyFraction()<0.80)
     return true;
   return false;
 }
+
+bool JetPFID::looseID2016_CHS(const Jet & jet) const{
+  if(fabs(jet.eta())<=2.7
+     && jet.numberOfDaughters()>1 
+     && jet.neutralHadronEnergyFraction()<0.99
+     && jet.neutralEmEnergyFraction()<0.99){
+    
+    if(fabs(jet.eta())>=2.4)
+      return true;
+      
+    if(jet.chargedEmEnergyFraction()<0.99
+       && jet.chargedHadronEnergyFraction()>0
+       && jet.chargedMultiplicity()>0)
+      return true;   
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  &&jet.neutralEmEnergyFraction()<0.90
```

Comment:
Hm looks like they updated the requirements for eta 2.7 - 3, now it's `jet.neutralEmEnergyFraction()>0.01 && jet.neutralHadronEnergyFraction()<0.98` instead.